x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/InlineJoin.java

```diff
@@ -139,20 +147,12 @@ public List<Attribute> computeOutput(List<Attribute> left, List<Attribute> right
         JoinType joinType = config().type();
         List<Attribute> output;
         if (LEFT.equals(joinType)) {
-            AttributeSet rightFields = AttributeSet.of(config().rightFields());
-            List<Attribute> leftOutputWithoutMatchFields = new ArrayList<>();
-            // at this point "left" part of the join contains all the attributes that represent the input of the join
-            // including any aliasing (evals) of expressions used as grouping attributes (or join "match fields") in the join itself
-            for (Attribute attr : left().output()) {
-                if (rightFields.contains(attr) == false) {
-                    // the aforementioned groupings expressions or aliasing are removed from the left set of attributes
-                    leftOutputWithoutMatchFields.add(attr);
-                }
-            }
-            // the actual output of the join will place the left hand side attributes (excluding any aliasing of the groupings)
-            // as first columns in the output followed by whatever the right hand side of join adds in this order: aggregates first,
-            // followed by groupings (this order should be preserved inside the rightFields() output)
-            output = mergeOutputAttributes(right, leftOutputWithoutMatchFields);
+            List<Attribute> leftOutputWithoutKeys = left.stream().filter(attr -> config().leftFields().contains(attr) == false).toList();
```

Comment:
This is taken from @alex-spies great suggestion [here](https://github.com/elastic/elasticsearch/pull/123589/files/b2c34742317c64c21a9b5019bd642cdb69178353#r1979692509)